pymatgen/core/surface.py

```diff
@@ -1231,8 +1228,7 @@ def gen_possible_shifts(ftol: float) -> list[float]:
 
             # Clustering does not work when there is only one atom
             if n_atoms == 1:
-                # TODO (@DanielYang59): why this magic number 0.5?
-                shift = frac_coords[0][2] + 0.5
+                shift = -frac_coords[0][2] + 0.5  # shift to center
```

Comment:
https://github.com/materialsproject/pymatgen/pull/3691#discussion_r1554518586 migrated here.

This looks like a typo, if the `shift` (in fractional coordinate) for single atom is `-frac_coords[0][2] + 0.5`, the atom would then be shifted to the center.

Reply:
did this change affect any tests?

Reply:
Yes it does, two tests were broken but I forgot exactly which one because [previous reversing of shifting direction breaks another two tests](https://github.com/materialsproject/pymatgen/pull/3748#discussion_r1561953247).

Even taking previous "positive shift value to move atoms downwards along z-axis" into account, this seems a typo to me.